build-tests/api-extractor-scenarios/etc/includeForgottenExports/api-extractor-scenarios.api.json

```diff
@@ -0,0 +1,638 @@
+              "canonicalReference": "api-extractor-scenarios!~ForgottenExport1:constructor(1)",
+              "docComment": "/**\n * Constructs a new instance of the `ForgottenExport1` class\n */\n",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "constructor();"
+                }
+              ],
+              "releaseTag": "None",
+              "isProtected": false,
+              "overloadIndex": 1,
+              "parameters": []
+            },
+            {
+              "kind": "Property",
+              "canonicalReference": "api-extractor-scenarios!~ForgottenExport1#prop:member",
+              "docComment": "",
+              "excerptTokens": [
+                {
+                  "kind": "Content",
+                  "text": "prop?: "
+                },
+                {
+                  "kind": "Reference",
+                  "text": "ForgottenExport2",
+                  "canonicalReference": "api-extractor-scenarios!~ForgottenExport2:type"
+                },
+                {
+                  "kind": "Content",
+                  "text": ";"
+                }
+              ],
+              "isReadonly": false,
+              "isOptional": true,
+              "releaseTag": "None",
+              "name": "prop",
+              "propertyTypeTokenRange": {
+                "startIndex": 1,
+                "endIndex": 2
+              },
+              "isStatic": false,
+              "isProtected": false
+            }
+          ],
+          "implementsTokenRanges": []
+        },
+        {
+          "kind": "TypeAlias",
+          "canonicalReference": "api-extractor-scenarios!~ForgottenExport2:type",
+          "docComment": "/**\n * {@inheritDoc ForgottenExport1}\n */\n",
+          "excerptTokens": [
+            {
+              "kind": "Content",
+              "text": "declare type ForgottenExport2 = "
+            },
+            {
+              "kind": "Content",
+              "text": "number"
+            },
+            {
+              "kind": "Content",
+              "text": ";"
+            }
+          ],
+          "releaseTag": "None",
+          "name": "ForgottenExport2",
+          "typeTokenRange": {
+            "startIndex": 1,
+            "endIndex": 2
+          }
+        },
+        {
+          "kind": "Namespace",
+          "canonicalReference": "api-extractor-scenarios!~ForgottenExport4:namespace",
+          "docComment": "",
+          "excerptTokens": [
+            {
+              "kind": "Content",
+              "text": "declare namespace ForgottenExport4 "
+            }
+          ],
+          "releaseTag": "None",
+          "name": "ForgottenExport4",
+          "preserveMemberOrder": false,
+          "members": [
+            {
+              "kind": "Class",
+              "canonicalReference": "api-extractor-scenarios!~ForgottenExport4~ForgottenExport5:class",
```

Comment:
This canonical reference should be `api-extractor-scenarios!~ForgottenExport4.ForgottenExport5:class`, because `ForgottenExport5` is exported from `ForgottenExport4`. However, unfortunately `ApiExportedMixin.isExported` is `false` for `ForgottenExport5`, because `ForgottenExport5` determines its exported state from `ForgottenExport4`'s `CollectorEntity` (i.e. it doesn't have its own `CollectorEntity`).

This is fundamentally an instance of the pre-existing bug 2 mentioned in the PR description.